clang/include/clang/CConv/DeclRewriter.h

```diff
@@ -57,15 +58,21 @@ class DeclRewriter {
   // Rewrite a specific variable declaration using the replacement string in the
   // DAndReplace structure. Each of these functions is specialized to handling
   // one subclass of declarations.
-  void rewriteVarDecl(VarDeclReplacement *N, RSet &ToRewrite);
   void rewriteParmVarDecl(ParmVarDeclReplacement *N);
+  template<typename D>
+  void rewriteMultiDecl(DeclReplacementTempl<D> *N, RSet &ToRewrite);
   void rewriteFunctionDecl(FunctionDeclReplacement *N);
 
-  SourceLocation deleteAllDeclarationsOnLine(VarDeclReplacement *N);
-  void getDeclsOnSameLine(VarDeclReplacement *N, std::set<Decl *> &Decls);
-  bool isSingleDeclaration(VarDeclReplacement *N);
-  bool areDeclarationsOnSameLine(VarDeclReplacement *N1,
-                                 VarDeclReplacement *N2);
+    template<typename D>
+  SourceLocation deleteAllDeclarationsOnLine(DeclReplacementTempl<D> *N);
```

Comment:
Can the abstract base class `DeclReplacement` be used instead of the template version?

Reply:
It makes the code in `rewriteMultiDecl` a little nicer, but removed the rest of them and used the ABC as it's cleaner.